packages/telemetry/shim.js

```diff
@@ -0,0 +1,66 @@
+/* global globalThis */
+import { MeterProvider } from '@opentelemetry/metrics';
+import { PrometheusExporter } from '@opentelemetry/exporter-prometheus';
+
+const configNames = [
+  'OTEL_EXPORTER_PROMETHEUS_HOST',
+  'OTEL_EXPORTER_PROMETHEUS_PORT',
+];
+
+const getConfig = env => {
+  const config = new Map();
+  configNames.forEach(name => {
+    const value = env[name];
+    if (value !== undefined) {
+      config.set(name, value);
+    }
+  });
+  return config;
+};
+
+// This is named getMetricProvider instead of getMeterProvider to avoid horrible
+// confusion with SwingSet metering.
+globalThis.getMetricProvider = (console, env) => {
+  const config = getConfig(env);
+
+  const getPrometheusMeterProvider = () => {
+    if (
+      !config.has('OTEL_EXPORTER_PROMETHEUS_HOST') &&
+      !config.has('OTEL_EXPORTER_PROMETHEUS_PORT')
+    ) {
+      // No Prometheus config, so don't install.
+      return undefined;
+    }
+
+    const host =
+      config.get('OTEL_EXPORTER_PROMETHEUS_HOST') ||
+      PrometheusExporter.DEFAULT_OPTIONS.host ||
+      '0.0.0.0';
+
+    const port =
+      Number(config.get('OTEL_EXPORTER_PROMETHEUS_PORT')) ||
+      PrometheusExporter.DEFAULT_OPTIONS.port;
```

Comment:
Might validate. `Number` can return `NaN`. Domain includes hex, if that’s intended to be supported. Not sure whether `0` should silently fall through to default it’s provided explicitly.

Reply:
Updated to use `parseInt(..., 10)`.